src/main/java/com/odysseusinc/arachne/datanode/service/impl/BaseCentralIntegrationServiceImpl.java

```diff
@@ -245,17 +211,22 @@ public User getUserInfoFromCentral(String centralToken) {
     }
 
     @Override
-    public JsonResult<CommonUserDTO> getUserFromCentral(User user, Long centralUserId) {
+    public JsonResult<CommonUserDTO> getUserFromCentral(User user, String username) {
 
-        return centralClient.getUser(centralUserId);
+        try {
+            return centralClient.getUser(URLEncoder.encode(username, "UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     @Override
     public void linkUserToDataNodeOnCentral(DataNode dataNode, User user) {
 
         final CommonLinkUserToDataNodeDTO commonLinkUserToDataNode
                 = conversionService.convert(user, CommonLinkUserToDataNodeDTO.class);
-        centralSystemClient.linkUser(dataNode.getCentralId(), commonLinkUserToDataNode);
+        JsonResult<CommonUserDTO> result = centralSystemClient.linkUser(dataNode.getCentralId(), commonLinkUserToDataNode);
+        user.setToken(result.getResult().getCentralToken());
```

Comment:
This seems to be related to impersonation, isn't it?

Reply:
Oh, it's just to mark user as synchronized with Central. 
One's useful to distinguish users who has created during the stand-alone mode.